hierarchy-tool/tool/src/java/org/sakaiproject/hierarchy/tool/entityproviders/HierarchyEntityProviderImpl.java

```diff
@@ -0,0 +1,101 @@
+package org.sakaiproject.hierarchy.tool.entityproviders;
+
+import lombok.Setter;
+import org.apache.commons.lang.StringUtils;
+import org.sakaiproject.entitybroker.EntityView;
+import org.sakaiproject.entitybroker.entityprovider.EntityProvider;
+import org.sakaiproject.entitybroker.entityprovider.annotations.EntityCustomAction;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.*;
+import org.sakaiproject.entitybroker.entityprovider.extension.Formats;
+import org.sakaiproject.entitybroker.util.AbstractEntityProvider;
+import org.sakaiproject.hierarchy.api.PortalHierarchyService;
+import org.sakaiproject.hierarchy.api.model.PortalNode;
+import org.sakaiproject.hierarchy.api.model.PortalNodeRedirect;
+import org.sakaiproject.hierarchy.api.model.PortalNodeSite;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Hierarchy entity used to access subsites and other details of a site
+ * Created by neelam on 15-Apr-16.
+ */
+public class HierarchyEntityProviderImpl extends AbstractEntityProvider
+		implements AutoRegisterEntityProvider, Describeable, EntityProvider, ActionsExecutable, Outputable, Sampleable {
+
+	public final static String ENTITY_PREFIX = "hierarchy";
```

Comment:
We should have a different prefix as there is a general hierarchy service which would make more sense to expose as hierarchy. Maybe `portal-hierarchy`?